File: app/src/main/java/com/hjc/reader/viewmodel/search/SearchViewModel.java
package com.hjc.reader.viewmodel.search;


import android.app.Application;

import androidx.annotation.NonNull;

import com.hjc.baselib.viewmodel.CommonViewModel;
import com.hjc.reader.bean.db.History;
import com.hjc.reader.bean.db.HistoryDao;
import com.hjc.reader.bean.db.HistoryDataBase;

import java.util.List;

public class SearchViewModel extends CommonViewModel {

    public SearchViewModel(@NonNull Application application) {
        super(application);
    }


    /**
     * 保存历史搜索数据
     *
     * @param keyword 搜索关键字
     */
    public void saveHistoryData(String keyword) {
        HistoryDao historyDao = HistoryDataBase.getInstance(getApplication()).getHistoryDao();

        List<History> historyList = historyDao.getAllHistory();
        boolean isExit = false;
        for (History entity : historyList) {
            if (keyword.equals(entity.getName())) {
                isExit = true;
                break;
            }
        }

        if (!isExit) {
            History history = new History();
            history.setName(keyword);
            historyDao.insert(history);
        }
    }
}

File: module-main/src/main/java/com/hjc/learn/main/viewmodel/search/SearchViewModel.kt
package com.hjc.learn.main.viewmodel.search

import android.app.Application
import com.hjc.learn.main.entity.History
import com.hjc.learn.main.entity.HistoryDao
import com.hjc.learn.main.entity.HistoryDataBase
import com.hjc.library_common.viewmodel.KotlinViewModel

/**
 * @Author: HJC
 * @Date: 2021/3/3 15:24
 * @Description: 搜索ViewModel
 */
class SearchViewModel(application: Application) : KotlinViewModel(application) {

    /**
     * 保存历史搜索数据
     *
     * @param keyword 搜索关键字
     */
    fun saveHistoryData(keyword: String) {
        val historyDao: HistoryDao = HistoryDataBase.getInstance(getApplication()).getHistoryDao()
        val historyList = historyDao.getAllHistory()
        var isExit = false
        historyList?.let {
            for (entity in it) {
                if (keyword == entity.name) {
                    isExit = true
                    break
                }
            }
            if (!isExit) {
                val history = History()
                history.name = keyword
                historyDao.insert(history)
            }
        }
    }
}